### PR TITLE
#2191 add knitr.child.warning option to hide warnings in case the there are child documents in a chunk and the chunk still contains code (which is not executed)

### PR DESCRIPTION
**Pull request by jwijffels**
Let me know if I need to add some tests or documentation and if so, where you want me to add it.

### CODE REVIEW

File: R/block.R [hunk 1]
```diff
@@ -56,7 +56,7 @@ call_block = function(block) {
   if (opts_knit$get('progress')) print(block)
 
   if (!is.null(params$child)) {
-    if (!is_blank(params[['code']])) warning(
+    if (!is_blank(params[['code']]) && getOption('knitr.child.warning', default = TRUE)) warning(
```
Comment (yihui) [inline R/block.R:59]:
```suggestion
    if (!is_blank(params[['code']]) && getOption('knitr.child.warning', TRUE)) warning(
```

File: inst/examples/child/knitr-main.Rmd [hunk 2]
```diff
@@ -1,4 +1,4 @@
-You can also use the `child` option to include child documents in markdown.
+You can also use the `child` option to include child documents in markdown.
```
Comment (yihui) [inline inst/examples/child/knitr-main.Rmd:1]:
```suggestion
You can also use the `child` option to include child documents in markdown.
```

**Review by yihui — APPROVED** [review]
Looks good. It will be great if this option is mentioned in `NEWS.md`. Thanks!

**Comment by jwijffels** [comment]
> Looks good. It will be great if this option is mentioned in `NEWS.md`. Thanks!

Done. If it needs to be refererred to somewhere else, let me know.

**Review by yihui — APPROVED** [review]
Looks perfect now. Thanks!

**Comment by jwijffels** [comment]
Thanks for your time!

**Pull request merged**